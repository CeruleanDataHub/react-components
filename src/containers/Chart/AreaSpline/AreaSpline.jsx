import React from "react";

import { Chart } from "../Chart";

/**
 * Use this component to generate Area Spline charts. Component properties are inherited from the `Chart` component.
 */
export const AreaSpline = ({ title, xAxis, series }) => (
  <Chart title={title} xAxis={xAxis} series={series} type="areaspline" />
);

AreaSpline.propTypes = {
  /** Title of the chart, type is inherited from the `Chart` base component */
  title: Chart.propTypes.title,
  /** Values for X axis, type is inherited from the `Chart` base component */
  xAxis: Chart.propTypes.xAxis,
  /** Data to be graphed, type is inherited from the `Chart` base component */
  series: Chart.propTypes.series
};

AreaSpline.defaultProps = {
  title: Chart.defaultProps.title,
  xAxis: Chart.defaultProps.xAxis,
  series: Chart.defaultProps.series
};
