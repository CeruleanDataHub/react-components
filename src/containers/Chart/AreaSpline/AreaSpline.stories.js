import React from "react";

import { AreaSpline } from "./AreaSpline";

export default {
  title: "AreaSpline",
  component: AreaSpline
};

const xAxis = [
  {
    categories: [
      "monday",
      "tuesday",
      "wednesday",
      "thursday",
      "friday",
      "saturday",
      "sunday"
    ]
  }
];

const series = [
  { name: "Temperature", data: [99, 80, 82, 59, 90, 6, 99] },
  { name: "Humidity", data: [29, 39, 26, 53, 9, 46, 31] }
];

export const AreaSplineStory = () => (
  <AreaSpline
    title="Sauna histogram as area spline"
    xAxis={xAxis}
    series={series}
  />
);
