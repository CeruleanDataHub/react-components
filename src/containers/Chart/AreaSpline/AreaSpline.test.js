import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import { AreaSpline } from "./AreaSpline";

describe("AreaSpline", () => {
  it("should render chart", () => {
    const component = shallow(<AreaSpline />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it("should render chart with simple data", () => {
    const xAxis = [
      {
        categories: [
          "monday",
          "tuesday",
          "wednesday",
          "thursday",
          "friday",
          "saturday",
          "sunday"
        ]
      }
    ];

    const series = [
      { name: "temperature", data: [10, 15, 20, 25, 15, 20, 15] }
    ];

    const component = shallow(
      <AreaSpline title="Chart" xAxis={xAxis} series={series} />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
