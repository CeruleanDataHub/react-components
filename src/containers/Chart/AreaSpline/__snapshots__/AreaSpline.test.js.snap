// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaSpline should render chart 1`] = `
<Chart
  options={Object {}}
  series={Array []}
  title=""
  type="areaspline"
  xAxis={null}
/>
`;

exports[`AreaSpline should render chart with simple data 1`] = `
<Chart
  options={Object {}}
  series={
    Array [
      Object {
        "data": Array [
          10,
          15,
          20,
          25,
          15,
          20,
          15,
        ],
        "name": "temperature",
      },
    ]
  }
  title="Chart"
  type="areaspline"
  xAxis={
    Array [
      Object {
        "categories": Array [
          "monday",
          "tuesday",
          "wednesday",
          "thursday",
          "friday",
          "saturday",
          "sunday",
        ],
      },
    ]
  }
/>
`;
